mlx/backend/metal/kernels/reduce.metal

```diff
@@ -28,7 +28,8 @@
 
 #define instantiate_reduce_helper_64b(inst_f, name, op) \
   inst_f(name, int64, int64_t, op)                      \
-  inst_f(name, uint64, uint64_t, op)
+  inst_f(name, uint64, uint64_t, op)                    \
+  inst_f(name, complex64, complex64_t, op)
```

Comment:
Thanks for adding support for `complex64`! It might be good to add tests here (or in the fuzzer is fine too if that's your plan) since we haven't tested that yet..